src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.HostStates.cs

```diff
@@ -15,15 +15,29 @@ internal partial class DiagnosticIncrementalAnalyzer
         private partial class StateManager
         {
             public IEnumerable<StateSet> GetAllHostStateSets()
-                => _hostAnalyzerStateMap.Values.SelectMany(v => v.OrderedStateSets);
```

Comment:
What about `Values.Where(...).SelectMany`?

Reply:
I think I like the loop form above.